src/chrome/local-runtime.ts

```diff
@@ -197,6 +205,36 @@ export default class LocalRuntime {
     }
   }
 
+  private async history(): Promise<NavigationHistory> {
+    const {Page} = this.client
+    await Promise.all([Page.enable()])
+    return await Page.getNavigationHistory()
+  }
+
+  private async back(): Promise<void> {
+    const history = await this.history()
+    if (history && history.currentIndex > 0) {
+      const {Page} = this.client
+      await Page.navigateToHistoryEntry({entryId: history.entries[history.currentIndex - 1].id})
+      await Page.loadEventFired()
```

Comment:
That's indeed a good point with the hash location. Another option might be [`Page.frameNavigated`](https://chromedevtools.github.io/devtools-protocol/tot/Page/#event-frameNavigated), we should try that out.
Regarding just evaluating `window.history.back()` - we try to stay as  "native" as possible to prevent side effects with the JS runtime, you never know what weird stuff people do in their global scope...